src/common/rbac/casbin.go

```diff
@@ -53,12 +57,86 @@ e = some(where (p.eft == allow)) && !some(where (p.eft == deny))
 m = g(r.sub, p.sub) && keyMatch2(r.obj, p.obj) && (r.act == p.act || p.act == '*')
 `
 
-// keyMatch2 determines whether key1 matches the pattern of key2, its behavior most likely the builtin KeyMatch2
-// except that the match of ("/project/1/robot", "/project/1") will return false
-func keyMatch2(key1 string, key2 string) bool {
-	key2 = strings.Replace(key2, "/*", "/.*", -1)
+type regexpEntry struct {
+	data      *regexp.Regexp
+	isAliveAt int64
```

Comment:
This is trivial as it's not exported, but why not use `time.Time` and call `isAliveAt` as `expiration`

Reply:
Changed to `time.Time` and renamed to `expiratedAt`

Reply:
regexEntry was removed because we force to purge the cache every 24 hours